aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/RetryPolicyDecorator.kt

```diff
@@ -31,6 +35,21 @@ class RetryPolicyDecorator : RustCodegenDecorator<ClientCodegenContext> {
 
     override fun supportsCodegenContext(clazz: Class<out CoreCodegenContext>): Boolean =
         clazz.isAssignableFrom(ClientCodegenContext::class.java)
+
+    override fun transformModel(service: ServiceShape, model: Model): Model {
+        // Allow downstream codegen that needs to know the retry type for a given operation
+        // to recreate it by passing the `runtimeConfig`
+        return ModelTransformer.create().mapShapes(model) { shape ->
+            if (shape is OperationShape) {
+                val retryType = { runtimeConfig: RuntimeConfig ->
+                    runtimeConfig.awsHttp().asType().member("retry::AwsErrorRetryPolicy")
```

Comment:
If this type doesn't change per operation, then why does it need to be specified per operation? Is the idea that there will be service-specific customization that only change some operations?

Reply:
I removed the need to do this by making retry policy a parameter of the fluent client generator